File: php/transform.php
<?php
include 'calculate.php';

function transform_data($data) {
    $transformed_data = array_filter(array_map(function($x) {
        if (!is_numeric($x)) {
            throw new Exception('Type error');
        }
        return $x * 1.1;
    }, $data), function($x) {
        return $x > 1;
    });
    return calculate_statistic($transformed_data);
}
?>

File: rust/src/transform.rs
use crate::calculate::calculate_statistic;

pub fn transform_data(data: &Vec<f64>) -> Result<f64, &'static str> {
    let transformed_data: Vec<f64> = data.iter()
        .map(|&x| x * 1.1)
        .filter(|&x| x > 1.0)
        .collect();
    calculate_statistic(&transformed_data)
}
